routes/user.js

```diff
@@ -577,6 +578,44 @@ router.post("/avatar", async function (req, res) {
     }
 });
 
+router.post("/birthday", async function (req, res){
+    res.setHeader("Content-Type", "application/json");
+    try{
+        let userId = await routeUtils.verifyLoggedIn(req);
+        let prop = String(req.body.prop);
+        var itemsOwned = await redis.getUserItemsOwned(userId);
+        var perm = "changeBday";
+
+        if (!(await routeUtils.verifyPermission(res, userId, perm))) {
+            return;
+        }
+        let value = String(req.body.date);
+
+        if (!itemsOwned.bdayChange) {
+            res.status(500);
+            res.send("You must purchase another birthday change from the Shop.");
+            return;
+        }
+
+        await models.User.updateOne(
+            { id: userId },
+            {
+                $set: { birthday: value, bdayChanged: true },
+                $inc: { "itemsOwned.bdayChange": -1 }
+            }
+        ).exec();
+        await redis.cacheUserInfo(userId, true);

```

Comment:
2023-03-20T13:47:17.255Z error: Cast to date failed for value "undefined" (type string) at path "birthday"
```
CastError: Cast to date failed for value "undefined" (type string) at path "birthday"
    at model.Query.exec (C:\Users\hui_j\beyondmaf\node_modules\mongoose\lib\query.js:4498:21)     
    at C:\Users\hui_j\beyondmaf\routes\user.js:606:11
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

Reply:
- Future fix